torch/csrc/jit/passes/onnx/fold_if_node.cpp

```diff
@@ -0,0 +1,203 @@
+#include <torch/csrc/jit/passes/onnx/fold_if_node.h>
+#include <torch/csrc/jit/passes/onnx/helper.h>
+#include <torch/torch.h>
+
+#include <c10/util/Optional.h>
+#include <algorithm>
+
+namespace torch {
+namespace jit {
+
+namespace onnx {
+using namespace ::c10::onnx;
+}
+
+static bool checkIfFold(Node* node, bool dynamic_axes) {
```

Comment:
try to think of a more general way (in loop or recursive) for both `checkIfFold` and `constantFoldingValue`.